GitUpKit/Interface/GIImageDiffView.m

```diff
@@ -0,0 +1,313 @@
+  NSBundle* bundle = NSBundle.gitUpKitBundle;
+  NSImage* patternImage = [bundle imageForResource:@"background_pattern"];
+  _checkerboardColor = [NSColor colorWithPatternImage:patternImage];
+  _transparencyCheckerboardLayer.backgroundColor = _checkerboardColor.CGColor;
+  [self.layer addSublayer:_transparencyCheckerboardLayer];
+
+  _currentImageView = [[NSImageView alloc] init];
+  _oldImageView = [[NSImageView alloc] init];
+  [self addSubview:_currentImageView];
+  [self addSubview:_oldImageView];
+
+  _oldImageMaskLayer = [[CALayer alloc] init];
+  _oldImageMaskLayer.backgroundColor = NSColor.blackColor.CGColor;
+  _oldImageView.wantsLayer = true;
+  _oldImageView.layer.mask = _oldImageMaskLayer;
+
+  _currentImageMaskLayer = [[CALayer alloc] init];
+  _currentImageMaskLayer.backgroundColor = NSColor.blackColor.CGColor;
+  _currentImageView.wantsLayer = true;
+  _currentImageView.layer.mask = _currentImageMaskLayer;
+
+  _dividerView = [[NSView alloc] init];
+  [self addSubview:_dividerView];
+
+  _progressIndicator = [[NSProgressIndicator alloc] initWithFrame:NSMakeRect(0, 0, 30, 30)];
+  _progressIndicator.style = NSProgressIndicatorStyleSpinning;
+  [self addSubview:_progressIndicator];
+  [_progressIndicator startAnimation:self];
+  _progressIndicator.hidden = true;
+
+  _panGestureRecognizer = [[NSPanGestureRecognizer alloc] initWithTarget:self action:@selector(didMoveSplit:)];
+  _clickGestureRecognizer = [[NSClickGestureRecognizer alloc] initWithTarget:self action:@selector(didMoveSplit:)];
+  [self addGestureRecognizer:_panGestureRecognizer];
+  [self addGestureRecognizer:_clickGestureRecognizer];
+}
+
+- (void)setDelta:(GCDiffDelta*)delta {
+  if (delta != _delta) {
+    _delta = delta;
+    [self updateCurrentImage];
+    [self updateOldImage];
+    self.percentage = 0.5;
+  }
+}
+
+- (void)setPercentage:(CGFloat)percentage {
+  _percentage = percentage;
+  [self setNeedsDisplay:true];
+}
+
+- (void)updateCurrentImage {
+  NSError* error;
+  NSString* newPath;
+  if (_delta.newFile.SHA1 != nil) {
+    newPath = [GILaunchServicesLocator.diffTemporaryDirectoryPath stringByAppendingPathComponent:_delta.newFile.SHA1];
+    NSString* newExtension = _delta.newFile.path.pathExtension;
+    if (newExtension.length) {
+      newPath = [newPath stringByAppendingPathExtension:newExtension];
+    }
+    if (![[NSFileManager defaultManager] fileExistsAtPath:newPath]) {
+      [self.repository exportBlobWithSHA1:_delta.newFile.SHA1 toPath:newPath error:&error];
+    }
+  } else {
+    newPath = [self.repository absolutePathForFile:_delta.canonicalPath];
+  }
+  _currentImageSize = [self imageSizeWithoutLoadingFromPath:newPath];
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    NSImage* limitedSizeImage = [self generateLimitedSizeImageFromPath:newPath];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      _currentImageView.image = limitedSizeImage;
+      [self setNeedsDisplay:true];
+    });
+  });
+}
```

Comment:
Hi @Stengo. Testing this out it seems quite performant and looks amazing! My only question is around how images appear when they are deleted. The modified view shows the split view, the additional view shows the full image of the image added, but (at least for me) the delete image view shows a split with the left half being the deleted and the right being an empty image. Show the logic here be that if there is no new image that we don't draw it? I would expect to show the full deleted image.